src/main/java/org/testng/DependencyMap.java

```diff
@@ -21,7 +22,20 @@
 
   public DependencyMap(ITestNGMethod[] methods) {
     for (ITestNGMethod m : methods) {
-    	m_dependencies.put( m.getRealClass().getName() + "." +  m.getMethodName(), m);
+      m_dependencies.put( m.getRealClass().getName() + "." +  m.getMethodName(), m);
```

Comment:
I would expect test methods to respect java override rules, i.e. `ClassB#methodB` is executed and `ClassA#methodB` is ignored. And as I see it currently works in testng in this way.
Anyway if you decide to forbid test methods override it should be clearly stated in the documentation and test error message.